platforms/core-configuration/declarative-dsl-core/src/main/resources/org/gradle/internal/declarativedsl/parsing/grammar.bnf

```diff
@@ -39,34 +39,28 @@ root ::= statements
 statement ::= (assignment | expression)
 statements ::= (statement semis?)*
 
-assignment ::= directlyAssignableExpression EQ expression {pin=2}
-expression ::= postfixUnaryExpression
+assignment ::= directlyAssignableExpression EQ primaryExpression {pin=2}
+expression ::= primaryExpression postfixUnarySuffixes
 
```

Comment:
Note for the future: we will have to reintroduce a branch taht resolves to `simpleIdentifier` when we add support for enums.